### PR TITLE
AGS: More commits from upstream

### PR DESCRIPTION
**Pull request by tag2015**
This PR continues the task to incorporate AGS upstream commits

from
https://github.com/adventuregamestudio/ags/commit/0f5c6178ac07b96f0703b35334d77f83977d43fd (not included)
up to
https://github.com/adventuregamestudio/ags/commit/17503d7bd066e8bd4a5af490c9ab053a47e4468f

Includes fix for trac [13882](https://bugs.scummvm.org/ticket/13882)

SKIPPED COMMITS:

All the commits related to the "touch-to-mouse emulation", since we still need to decide if it's needed
0f5c6178ac07b96f0703b35334d77f83977d43fd
571d1ccfbb560e31998088e5efe00b415dd572d3
1d59972ba2e4560dbc2c327f7ffdfada9815f938
f9a5ee81a34acfcd63137703df10d199e2eb0013
7d33416a9c96137ec5bc0c738751de0434394d04
14f34b60067679064208096538010702192560b7
5f5acede172c0162e5a721bc740f32d5be1b589f
cc3f2a45f56db9d305e21c82ce4589c76968c944

Engine: fix agsblend plugin stub missing DrawSprite
f3970d07138cd71172070aeca7ad6c326e389d17
Not needed

Engine: in OpenAlSource::GetPositionMs return prediction if no buffers 
084f3f281359560a94ed819ff649edbf8bf9c4a0
Scummvm has its own sound code

Common: fixed ReadEncInt32() on big-endian systems
5e29a339fc83bf5c06a3a9a3b1c65a2fc4b4e72c
already merged

Common: small fix to 5e29a33
427752da015fd93549deef1a31d5e533e5c9319e
already merged

Engine: fix gfx sw driver presented offset twice
6fc888e04c4130fb4f58bc2608438c4057cb0701
Blitting code is different, should be fixed already

Engine: support alternate name for stub/builtin spritefont plugin
1f258dec6875910b385ed1ef46b7be7c4b9ed928
Already in scummvm

SpriteFont: add "The Excavation at Hob's Barrow" to Clifftop detection
fe218e52b99b73819e9ff3ba4e6b999b42642b3d
Different implementation, already in scummvm

SpriteFont: fixed a typecast warning
a016cd18794c0baba1763f2a43586ed472ea3550
Not needed

SpriteFont: support explicit choice between variants using cflags
9f5244ad6c3c07b4e1a6c3623d2f508ec93b45f6
Different implementation

Engine: fix freebsd comment
259a376bcae031ed099cc881b5a2892487f9f090
Not present

OSX: enable RMB simulation with Ctrl+LMB (in SDL2)
3fd5531f79044f58c261d9af0c7401ab1edf4ca7
PENDING: is this needed?

### CODE REVIEW

File: engines/ags/engine/ac/display.cpp
Comment (criezy) [inline engines/ags/engine/ac/display.cpp]: It's not going to break anything, but the upstream commit also change the above line to use a `const int`.

Comment (criezy) [reply engines/ags/engine/ac/display.cpp]: Ah. I see you actually changed it in the next commit. 😄 
So it's fine.

**Review by criezy — APPROVED** [review]
Thank you for this work. It is all good and I will merge it now.

> OSX: enable RMB simulation with Ctrl+LMB (in SDL2)
> 3fd5531f79044f58c261d9af0c7401ab1edf4ca7
> PENDING: is this needed?

This should be merged as well. This only removes unused code (since `AGS_SIMULATE_RIGHT_CLICK__FIXME` is never defined), so will not changed anything. But for consistency with upstream and to avoid wondering why that code is here in the future, it's better to remove it from ScummVM as well.
However this can be done directly in tree after this PR is merged.

**Comment by criezy** [comment]
To clarify my last comment about the OSX commit, only the change in `Engine/ac/sys_events.cpp` should be merged.

The second part of the commit (calling `SDL_SetHint(SDL_HINT_MAC_CTRL_CLICK_EMULATE_RIGHT_CLICK, "1");`) is something unrelated to the AGS engine and that I need to consider for the OSX backend in ScummVM.

**Comment by tag2015** [comment]
> To clarify my last comment about the OSX commit, only the change in `Engine/ac/sys_events.cpp` should be merged.
> 
> The second part of the commit (calling `SDL_SetHint(SDL_HINT_MAC_CTRL_CLICK_EMULATE_RIGHT_CLICK, "1");`) is something unrelated to the AGS engine and that I need to consider for the OSX backend in ScummVM.
Ok, I pushed the commit right now. Thanks for reviewing!

**Pull request merged**
